azure-core/azure/core/pipeline/transport/aiohttp.py

```diff
@@ -99,16 +107,13 @@ def _get_request_data(self, request):
         Will pre-load the body into memory to be available with a sync method.
         pass stream=True to avoid this behavior.
         """
-        if self.config.connection.keep_alive and self.session:
-            session = self.session
-        else:
-            session = aiohttp.ClientSession(loop=self._loop)
+        await self.open()
         error = None
         response = None
         config['ssl'] = self._build_ssl_config(**config)
         try:
             stream_response = config.pop("stream", False)
```

Comment:
Do we still need it?

Reply:
Yes - we still need it to determine whether to load the response body on line 127